base/number.jl

```diff
@@ -110,3 +110,27 @@ but you can use `factorial(big(n))` to compute the result exactly in arbitrary p
 If `n` is not an `Integer`, `factorial(n)` is equivalent to [`gamma(n+1)`](:func:`gamma(n+1) <gamma>`).
 """
 factorial(x::Number) = gamma(x + 1) # fallback for x not Integer
+
+"""
+    rescale(x, from_min, from_max, to_min=0.0, newmax=1.0)
+
+Convert `x` from one linear scale (`from_min` to `from_max`) to another (`to_min` to `newmax`).
+If unspecified, the target scale is 0.0 to 1.0.
+
+```jldoctest
+julia> rescale(15, 0, 100, 0, 1)
+0.15
+
+julia> rescale(pi/20, 0, 2pi)
+0.025
+
+julia> rescale(25, 0, 1, 0, 1.609344)
+40.2336
+
+julia> rescale(15, 0, 100, 1000, 0)
+850.0
+```
+"""
+
+rescale(x, from_min, from_max, to_min=0.0, newmax=1.0) =
```

Comment:
I would suggest either not including defaults for `to_min` and `to_max` (my preference) or using 0 and 1 rather than 0.0 and 1.0. That might make it "safer" from a type point of view, since 0.0 and 1.0 are `Float64`, which makes things unnecessarily promote to `Float64`.